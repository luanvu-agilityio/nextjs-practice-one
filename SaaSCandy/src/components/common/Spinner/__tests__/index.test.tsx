import { render } from '@testing-library/react';
import { Spinner } from '../index';

describe('Spinner', () => {
  it('matches snapshot', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });

  it('has loading role and aria-label', () => {
    const { getByRole } = render(<Spinner />);
    const spinner = getByRole('status');
    expect(spinner).toHaveAttribute('aria-label', 'Loading');
  });

  it('applies custom className', () => {
    const { getByRole } = render(<Spinner className='custom-spinner' />);
    expect(getByRole('status')).toHaveClass('custom-spinner');
  });
});
