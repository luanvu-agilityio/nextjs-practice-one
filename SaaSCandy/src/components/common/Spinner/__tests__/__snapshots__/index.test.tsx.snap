// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Spinner > matches snapshot 1`] = `
<div>
  <svg
    aria-label="Loading"
    class="lucide lucide-loader-circle size-4 animate-spin"
    fill="none"
    height="24"
    role="status"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 12a9 9 0 1 1-6.219-8.56"
    />
  </svg>
</div>
`;

exports[`Spinner matches snapshot 1`] = `
<div>
  <svg
    aria-label="Loading"
    class="lucide lucide-loader-circle size-4 animate-spin"
    fill="none"
    height="24"
    role="status"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 12a9 9 0 1 1-6.219-8.56"
    />
  </svg>
</div>
`;
